packages/middleware-stack/src/MiddlewareStack.ts

```diff
@@ -1,381 +1,228 @@
-   *
-   * This function normalizes relative middleware into 2 categories of linked
-   * lists. (1) linked list that have absolute-located middleware on one end.
-   * These middleware will be resolved accordingly before or after the absolute-
-   * located middleware. (2) Linked list that have no absolute-located middleware
-   * on any end. They will be resolved to corresponding step with normal priority
-   *
-   * The 2 types of linked list will return as a tuple
+   * Get a final list of middleware in the order of being executed in the resolved handler.
    */
-  private normalizeRelativeEntries(): NormalizingEntryResult<Input, Output> {
-    const absoluteMiddlewareNamesMap = this.absoluteEntries
-      .filter((entry) => entry.name)
-      .reduce((accumulator, entry) => {
-        accumulator[entry.name!] = entry;
-        return accumulator;
-      }, {} as NamedMiddlewareEntriesMap<Input, Output>);
-    const normalized = this.relativeEntries.map(
-      (entry) =>
-        ({
-          ...entry,
-          priority: null,
-          next: undefined,
-          prev: undefined,
-        } as NormalizedRelativeEntry<Input, Output>)
-    );
-    const relativeMiddlewareNamesMap = normalized
-      .filter((entry) => entry.name)
-      .reduce((accumulator, entry) => {
-        accumulator[entry.name!] = entry;
-        return accumulator;
-      }, {} as NamedRelativeEntriesMap<Input, Output>);
-
-    const anchors: RelativeMiddlewareAnchor<Input, Output> = {};
-    for (let i = 0; i < this.relativeEntries.length; i++) {
-      const { prev, next } = this.relativeEntries[i];
-      const resolvedCurr = normalized[i];
-      //either prev or next is set
-      if (prev) {
-        if (absoluteMiddlewareNamesMap[prev] && absoluteMiddlewareNamesMap[prev].step === resolvedCurr.step) {
-          if (!anchors[prev]) anchors[prev] = {};
-          resolvedCurr.next = anchors[prev].next;
-          if (anchors[prev].next) anchors[prev].next!.prev = resolvedCurr;
-          anchors[prev].next = resolvedCurr;
-        } else if (relativeMiddlewareNamesMap[prev] && relativeMiddlewareNamesMap[prev].step === resolvedCurr.step) {
-          const resolvedPrev = relativeMiddlewareNamesMap[prev];
-          if (resolvedPrev.next === resolvedCurr) continue;
-          resolvedCurr.next = resolvedPrev.next;
-          resolvedPrev.next = resolvedCurr;
-          if (resolvedCurr.next) resolvedCurr.next.prev = resolvedCurr;
-          resolvedCurr.prev = resolvedPrev;
+  const getMiddlewareList = (): Array<MiddlewareType<Input, Output>> => {
+    const normalizedAbsoluteEntries: Normalized<AbsoluteMiddlewareEntry<Input, Output>, Input, Output>[] = [];
+    const normalizedRelativeEntries: Normalized<RelativeMiddlewareEntry<Input, Output>, Input, Output>[] = [];
+    const normalizedEntriesNameMap: {
+      [middlewareName: string]: Normalized<MiddlewareEntry<Input, Output>, Input, Output>;
+    } = {};
+    absoluteEntries.forEach((entry) => {
```

Comment:
newline for better readability
```suggestion

    absoluteEntries.forEach((entry) => {
```

Ditto on lines 130, 139, 157 and other places.